tephra-core/src/main/java/org/apache/tephra/distributed/TransactionService.java

```diff
@@ -42,28 +45,64 @@
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
+import javax.annotation.Nullable;
 
 /**
  *
  */
-public final class TransactionService extends InMemoryTransactionService {
+public class TransactionService extends AbstractService {
```

Comment:
Add a javadoc about this class to tell what does it do and when it should be used.